core/document/src/layers/folder.rs

```diff
@@ -65,6 +65,84 @@ impl Folder {
 		Ok(())
 	}
 
+	pub fn reorder_layers(&mut self, source_ids: Vec<LayerId>, target_id: LayerId) -> Result<(), DocumentError> {
+		let source_pos = self.layer_ids.iter().position(|x| *x == source_ids[0]).ok_or(DocumentError::LayerNotFound)?;
+		let source_pos_end = source_pos + source_ids.len() - 1;
+		let target_pos = self.layer_ids.iter().position(|x| *x == target_id).ok_or(DocumentError::LayerNotFound)?;
+
+		let mut last_pos = source_pos;
+		for layer_id in &source_ids[1..source_ids.len()] {
```

Comment:
can you do something like `source_ids.take(source_ids.len())`? might be a bit more ideomatic

Reply:
That isn't equivalent code, the slice excludes the first element intentionally. If there's a better way of doing that I'm all ears.

Reply:
I suppose you could call `skip(1)` but at that point it might get longer than your current code.
Oh, wait, you are indexing source ids with source ids, in that case it should either be `source_ids.skip(1)` or &source_ids[1..]

Reply:
The nice thing about the skip is, that it does not panic if the list is empty but that depends on what you want the code to do

Reply:
Ah I see. The caller definitely shouldn't be passing in an empty list (moving a list of zero layers makes no sense).

I think `&source_ids[1..]` is probably the best solution here.